private/bufpkg/bufcheck/rules_config.go

```diff
@@ -102,6 +104,10 @@ type rulesConfig struct {
 	// Rules, we don't warn for this plugin, as it may have had rules configured in breaking that
 	// we haven't accounted for.
 	//
+	// The caller can provider additional check configs to check if the plugin has rules configured.
+	// If the plugin has rules configured in any of the additional check configs provided, then
+	// we don't warn.
```

Comment:
The case above (`In client.Lint and client.Breaking, we only have the Lint or Breaking config`) could probably be removed now to say related configs are provided for validation and remove the heuristic workaround based on other rule types.

Reply:
So this is one I was kinda hoping to get opinions on. Currently, I am only providing the respective other check configs (e.g. other lint configs for lint, other breaking configs for breaking), but yes, we could provide both instead, and that would allow us to remove this clause and rely solely on the related check configs. I like that idea, but I wanted to get others' opinions on this.

Reply:
I thought a little bit more about this, and the goal of this is to provide users with a clear warning for whenever a plugin does not have any configured rules and/or categories. And so I think it actually doesn't make sense to rely on the heuristic, it seems we are doing this because `RulesConfig` always only ever applies to a single `RuleType`:

https://github.com/bufbuild/buf/blob/b4d78361a823ccad7b204bbd7e320fac4df253cc/private/bufpkg/bufcheck/rules_config.go#L126

However, at the time of the validation, we have access to all rules, and so we can check all related check configs, agnostic to rule type, which would allow us to surface a much clearer warning to users without relying on a heuristic. So I have pushed up a change to reflect that for the warning.